File: plugins/devkit/testData/inspections/statefulEp/ProjectComp.java
public class ProjectComp implements com.intellij.openapi.components.ProjectComponent {
  <warning descr="Potential memory leak: don't hold PsiElement, use SmartPsiElementPointer instead">final com.intellij.psi.PsiElement pe;</warning>
  <warning descr="Don't use PsiReference as a field in extension">final com.intellij.psi.PsiReference r;</warning>
  com.intellij.openapi.project.Project p;
  final com.intellij.openapi.project.Project pf;
  public ProjectComp() {
    super();
    pe = null;
    r =null;
    p = pf = null;
  }
}

File: plugins/devkit/devkit-kotlin-tests/testData/inspections/statefulEp/ProjectComp.kt
import com.intellij.openapi.components.ProjectComponent
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiElement
import com.intellij.psi.PsiReference

class ProjectComp : ProjectComponent {
  val <warning descr="Potential memory leak: don't hold PsiElement, use SmartPsiElementPointer instead">pe</warning>: PsiElement?
  val <warning descr="Do not use PsiReference as a field in extension">r</warning>: PsiReference?
  var p: Project?
  val pf: Project?

  init {
    pe = null
    r = null
    pf = null
    p = pf
  }
}
